lib/valueflow.cpp

```diff
@@ -944,7 +944,7 @@ static void valueFlowRightShift(TokenList& tokenList, const Settings& settings)
         if (!tok->astOperand2()->hasKnownValue())
             continue;
 
-        const MathLib::bigint rhsvalue = tok->astOperand2()->values().front().intvalue;
+        const MathLib::bigint rhsvalue = tok->astOperand2()->getKnownValue()->intvalue;
```

Comment:
Why not getKbownIntValue?

Reply:
Because the preceding check is `hasKnownValue()`.

Reply:
That looks like a bug. It should be checking for a known int value.

Reply:
That is a bug, it should be checking for a known int value since it's creating an int value below.

Reply:
I see.

But the value is created on `tok` which has *another* `hasKnownValue()` check above. So that does not *seem* related (to me). Unless all of them need to be adjusted.

Also given the information in the other comment if this already has a non-int value and we set the int-value wouldn't that break that the case that it may only hold a single value (if that is even possible)?

Reply:
It is related, all the checks in this pass should check for int value only.